pyvista/core/utilities/geometric_objects.py

```diff
@@ -39,6 +41,247 @@
 }
 
 
+@no_new_attr
+class ConeSource(_vtk.vtkConeSource):
+    """Create a cone source algorism class.
+
+    Parameters
+    ----------
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``. Axis of the cone passes through this
+        point.
+
+    direction : sequence[float], default: (1.0, 0.0, 0.0)
+        Direction vector in ``[x, y, z]``. Orientation vector of the
+        cone.
+
+    height : float, default: 1.0
+        Height along the cone in its specified direction.
+
+    radius : float, optional
+        Base radius of the cone.
+
+    capping : bool, optional
+        Enable or disable the capping the base of the cone with a
+        polygon.
+
+    angle : float, optional
+        The angle in degrees between the axis of the cone and a
+        generatrix.
+
+    resolution : int, default: 6
+        Number of facets used to represent the cone.
+
+    Examples
+    --------
+    Create a default ConeSource.
+
+    >>> import pyvista
+    >>> source = pyvista.ConeSource()
+    >>> source.output.plot(show_edges=True, line_width=5)
+    """
+
+    def __init__(
+        self,
+        center=(0.0, 0.0, 0.0),
+        direction=(1.0, 0.0, 0.0),
+        height=1.0,
+        radius=None,
+        capping=True,
+        angle=None,
+        resolution=6,
+    ):
+        """Initialize the cone source class."""
+        super().__init__()
+        self.center = center
+        self.direction = direction
+        self.height = height
+        self.capping = capping
+        if angle and radius:
```

Comment:
Angle or radius of 0 will evaluate to False, even though it is a defined angle.

Reply:
Good catch! Thanks.

Reply:
resolved in fe9ae0d09b5517313106f9b14c5aa637e78c08d9